src/__experimental__/components/numeral-date/numeral-date.d.ts

```diff
@@ -1,5 +1,17 @@
 import * as React from 'react';
 
+interface NumeralDateEvent {
+  target: {
+    name: string,
+    id: string,
+    value: {
+     dd: string,
+     mm: string,
+     yyyy?: string,
+   },
+  };
+}
```

Comment:
we have multiple allowed formats that could be passed as that value:
```
  Allowed formats:
  ['dd', 'mm', 'yyyy'],
  ['mm', 'dd', 'yyyy'],
  ['dd', 'mm'],
  ['mm', 'dd'],
  ['mm', 'yyyy']
```

we need to add interfaces for them

```suggestion
interface DayMonthDate {
   dd: string;
   mm: string;
}

interface MonthYearDate {
   mm: string;
   yyyy: string;
}

interface FullDate extends DayMonthDate {
   yyyy: string;
}

interface NumeralDateEvent {
  target: {
    name: string,
    id: string,
    value: DayMonthDate | MonthYearDate |  FullDate,
  };
}
```